engine/apps/grafana_plugin/tasks/sync_v2.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import math
+from time import sleep
+
+from celery.utils.log import get_task_logger
+from django.utils import timezone
+
+from apps.grafana_plugin.helpers.client import GrafanaAPIClient
+from apps.grafana_plugin.helpers.gcom import get_active_instance_ids
+from apps.user_management.models import Organization
+from common.custom_celery_tasks import shared_dedicated_queue_retry_task
+
+logger = get_task_logger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+SYNC_PERIOD = timezone.timedelta(minutes=4)
+
+
+@shared_dedicated_queue_retry_task(autoretry_for=(Exception,), retry_backoff=True, max_retries=0)
+def sync_organizations_v2():
+    organization_qs = Organization.objects.all()
+    active_instance_ids, is_cloud_configured = get_active_instance_ids()
+    if is_cloud_configured:
+        if not active_instance_ids:
+            logger.warning("Did not find any active instances!")
+            return
+        else:
+            logger.debug(f"Found {len(active_instance_ids)} active instances")
+            organization_qs = organization_qs.filter(stack_id__in=active_instance_ids)
+
+    orgs_per_second = math.ceil(len(organization_qs) / SYNC_PERIOD.seconds)
+    for idx, org in enumerate(organization_qs):
+        client = GrafanaAPIClient(api_url=org.grafana_url, api_token=org.api_token)
+        _, status = client.sync()
+        if status["status_code"] != 200:
+            logger.error(
+                f"Failed to request sync stack_slug={org.stack_slug} status_code={status['status_code']} url={status['url']} message={status['message']}"
+            )
+        if idx % orgs_per_second == 0:
+            logger.info(f"Sleep 1s after {idx + 1} organizations processed")
+            sleep(1)
```

Comment:
Does this mean we will have a ~4min running task every 5mins? Will that keep one worker busy most of the time?

Reply:
Yeah the thinking here was to avoid having the task slam requests as fast as possible. Not sure it is worth doing or we should try to get through them as fast as possible.  Another option could be running a few workers concurrently which handle different hashes of ids to divide the work. wdyt?

Reply:
+1 to spread the updates. I guess we can see how this behaves and have a plan in case we need to tune settings if needed? Also we may need to keep track of the timing of sync requests to know how this could affect latency too? Eventually if we could skip no-op syncs would be great (and/or apply "diffs" as we also discussed).